crates/net/src/http/headers.rs

```diff
@@ -6,6 +6,7 @@ use wasm_bindgen::{JsCast, UnwrapThrowExt};
 // I experimented with using `js_sys::Object` for the headers, since this object is marked
 // experimental in MDN. However it's in the fetch spec, and it's necessary for appending headers.
 /// A wrapper around `web_sys::Headers`.
+#[derive(Clone, PartialEq, Eq)]
```

Comment:
```rust
let headers = todo!()
let headers2 = headers.clone()
headers2.append("key123", "value")

assert!(!headers.has("key123"))
```
Can you check that this above snippet doesn't panic? I want to ensure that we're not sharing the same instance of `web_sys::Headers`

Reply:
I agree with @futursolo that derives here are most likely unsound. Now I understand the `web_sys` `Clone, PartialEq, Eq` implementations call out to JS to preform the shallow copies and comparisons